tests/unit/locking_test.cc

```diff
@@ -420,3 +420,171 @@ SEASTAR_THREAD_TEST_CASE(test_with_lock_typed_return_throwing_move_func) {
         }
     }
 }
+
+SEASTAR_THREAD_TEST_CASE(test_shared_mutex_locks) {
+    shared_mutex sm;
+
+    {
+        const auto ulock = get_unique_lock(sm).get();
+        BOOST_REQUIRE(!sm.try_lock());
+        BOOST_REQUIRE(!sm.try_lock_shared());
+    }
+
+    BOOST_REQUIRE(sm.try_lock());
+    sm.unlock();
+
+    {
+        const auto slock1 = get_shared_lock(sm).get();
+        BOOST_REQUIRE(!sm.try_lock());
+        BOOST_REQUIRE(sm.try_lock_shared());
+
+        const auto slock2 = get_shared_lock(sm).get();
+        sm.unlock_shared();
```

Comment:
please add comment that this balances the successful `try_lock_shared()` call above